src/Domain/Regulation/Repository/LocationRepositoryInterface.php

```diff
@@ -11,4 +11,8 @@ interface LocationRepositoryInterface
     public function add(Location $location): Location;
 
     public function delete(Location $location): void;
+
+    public function findOneByUuid(string $uuid): ?Location;
+
+    public function findFilteredLocationsAsGeoJson(string $permanentAndOrTemporaryFilter, string $futureFilter, string $pastFilter): array;
```

Comment:
Le naming des variables est à affiner je pense

Le mot "filter" n'est pas clair je trouve, il vaut mieux utiliser un nommage comme "include ceci, exclude cela", en évitant les variables tri-état dont la définition ne peut pas être faite dans cette méthode (à moins de définir un enum) et est donc difficile à maintenir

Enfin, les géométries GeoJSON côté PHP devraient toujours être des strings car on doit minimiser le traitement de GeoJSON en PHP (il vaut mieux passer par PostGIS quand on en a besoin)

Du coup j'imaginerais plutôt qqc comme ça...

```suggestion
    public function findAllForMapAsGeoJSON(bool $includePermanent = true, bool $includeTemporary = true, bool $upcomingOnly = false): string;
```